xdp-loader/tests/test-xdp-loader.sh

```diff
@@ -13,6 +13,52 @@ test_section()
     check_run $XDP_LOADER unload $NS --all -vv
 }
 
+check_progs_loaded()
+{
+    local iface="$1"
+    local num=$2
+    local num_loaded
+
+    num_loaded=$($XDP_LOADER status $NS | grep -c '=>')
+    if [ "$num_loaded" -ne "$num" ]; then
+        echo "Expected" $num "programs loaded, found" $num_loaded
+        exit 1
+    fi
+}
+
+test_load_multi()
+{
+    check_run $XDP_LOADER load $NS $TEST_PROG_DIR/xdp_drop.o $TEST_PROG_DIR/xdp_pass.o -vv
+    check_progs_loaded $NS 2
+    check_run $XDP_LOADER unload $NS --all -vv
+}
+
+test_load_incremental()
+{
+    local output
+    local ret
+    local id
+
+    check_run $XDP_LOADER load $NS $TEST_PROG_DIR/xdp_drop.o -vv
+
+    check_progs_loaded $NS 1
+
+    output=$($XDP_LOADER load $NS $TEST_PROG_DIR/xdp_pass.o -vv)
+    ret=$?
+
+    if [ "$ret" -ne "0" ] && echo $output | grep -q "Falling back to loading single prog"; then
+        ret=$SKIPPED_TEST
+    else
+        check_progs_loaded $NS 2
+
+        id=$($XDP_LOADER status $NS | grep xdp_pass | awk '{print $4}')
+        check_run $XDP_LOADER unload $NS --id $id
+        check_progs_loaded $NS 1
```

Comment:
I think the following needs to be added, so we test the unload all corner case trough IDs:

```
        id=$($XDP_LOADER status $NS | grep xdp_drop | awk '{print $4}')                                                                                                                                                                                         
        check_run $XDP_LOADER unload $NS --id $id                                                                                                                                                                                                               
        check_progs_loaded $NS 0  
```

For now it's failing as check_progs_loaded() needs fixing.